### PR TITLE
Fix lots of bugs that occur when playing multiple matches

### PR DESCRIPTION
**Pull request by msoechting**
Please test this locally!! Leave matches during games early, in warmup, etc. and try to break it. It's pretty solid by now (lots of guards required).

### CODE REVIEW

File: src/tetris/scene/playScene.ts [hunk 1]
```diff
@@ -219,10 +235,13 @@ export default class PlayScene extends Phaser.Scene {
 	}
 
 	private _exitMatch(): void {
-		this._game.handleEndOfMatch(this._match);
+		this._game.changeScene(config.sceneKeys.menuScene);
```
Comment (julianweise) [inline src/tetris/scene/playScene.ts]: Usually, changing scene took place in `handleEndOfMatch()`.

Comment (msoechting) [reply src/tetris/scene/playScene.ts]: In current version of the developer, there's no such call. The only subscriber to the endOfMatch event is the profiler, which definitely doesnt change the scene.

Comment (julianweise) [reply src/tetris/scene/playScene.ts]: Ahh sorry, mixed it up with the ExitButton (exitGame)

Comment (msoechting) [reply src/tetris/scene/playScene.ts]: 👌

**Pull request merged**